addons/website_sale/controllers/main.py

```diff
@@ -1054,25 +1054,40 @@ def add_product(self, name=None, category=None, **post):
         })
         return "%s?enable_editor=1" % product.product_tmpl_id.website_url
 
+    @http.route(['/shop/add_or_edit_style'], type='json', auth='user', methods=['POST'], website=True)
+    def add_or_edit_style(self, name, color, id=None, edit=False):
+        product_style = request.env['product.style']
+        html_class = 'oe_ribbon_' + name.replace(' ', '_')
+        if edit:
+            product_style.browse(id).write({
+                'name': name,
+                'html_class': html_class,
+                'color': color
+            })
+            return id
+        else:
+            new_ribbon = product_style.create({
+                'name': name,
+                'html_class': html_class,
+                'color': color
+            })
+            return new_ribbon.id
+
     @http.route(['/shop/change_styles'], type='json', auth='user')
     def change_styles(self, id, style_id):
         product = request.env['product.template'].browse(id)
 
         remove = []
         active = False
         style_id = int(style_id)
-        for style in product.website_style_ids:
-            if style.id == style_id:
-                remove.append(style.id)
-                active = True
-                break
+        if product.website_style_id.id == style_id:
+            product.write({'website_style_id': False})
+            active = True
 
         style = request.env['product.style'].browse(style_id)
 
-        if remove:
-            product.write({'website_style_ids': [(3, rid) for rid in remove]})
         if not active:
-            product.write({'website_style_ids': [(4, style.id)]})
+            product.write({'website_style_id': style.id})
```

Comment:
==>
:/

`remove = [] `-> not more used
`style = request.env['product.style'].browse(style_id) `-> useless, you just use the id later...
`if ...:     val = True if not val:     xxx`  => or just an if / else 

=> 

```python
new_value = False if product.website_style_id.id == style_id else style_id
product.write({'website_style_id': new_value})
```